language: python
sudo: false
python:
  - "2.7"
  - "3.3"
  - "3.4"
env:
  global:
    - secure: "fHTmXSkUyVFu7lJ4upcu4Xhv+5cTGq+SJ+GV9UUJ9guZYVWafxJw3CRmd/qqjucQzOTPg/2JA4/9XqDZDSWKKBwGTOgeiFjKaa4xPFO/bvpbECqR4y7TDh7k7zSjMCDD624b8k6LyWiaerh50ShU2CSi3cTjpyZWloolQWRotLM="
    - REPO: "ASPP/pelita"
  matrix:
    - PYZMQ: "pyzmq"
    - PYZMQ: "pyzmq==2.2.0.1"
# command to install dependencies
install:
  - pip install six
  - pip install $PYZMQ --install-option="--zmq=bundled" --use-mirrors
# command to run tests
script: >
          python -c "from __future__ import print_function; import zmq; print('Using zmq %s.' % zmq.zmq_version())" &&
          python -c "from __future__ import print_function; import zmq; print('Using pyzmq %s.' % zmq.pyzmq_version())" &&
          nosetests -v && ./pelitagame --progress &&
          for player in ./players/*Player.py
            do
              PYTHONPATH=.
              ./pelitagame --check-team $player || exit 1
              res=$(./pelitagame --null --rounds 100 --filter small $player 2>&1)
              echo $res
              (echo $res | grep -q Exception) && exit 1
              echo "" # This resets the error code
          done
after_success:
  - ./deploy-gh-pages.sh
